hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -60,14 +61,39 @@ public HoodieCommitArchiveLog(HoodieWriteConfig config, FileSystem fs) {
      */
     public boolean archiveIfRequired() {
         List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
+        boolean success = true;
         if (commitsToArchive.iterator().hasNext()) {
             log.info("Archiving commits " + commitsToArchive);
             archive(commitsToArchive);
-            return deleteCommits(commitsToArchive);
+            success = deleteInstants(commitsToArchive);
         } else {
             log.info("No Commits to archive");
-            return true;
         }
+        return success & deleteOtherInstants();
+    }
+
+    private boolean deleteOtherInstants() {
```

Comment:
why not archive the timeline as-it-is? I am a little concerned about the special casing here..  

Conceptually, timeline contains (inflight, completed) instants of different actions and we archive it as it is, including cleaning too right? (same comment I had on the other PR about .clean.inflight).